code/game/smoothwall.dm

```diff
@@ -77,6 +77,17 @@
 				if(isSmoothableNeighbor(A))
 					A.relativewall()
 
+/atom/proc/update_near_walls(var/at)
```

Comment:
What was wrong with the proc directly above that this was copied and pasted from?

Reply:
This proc does not care if the atom calling it is a wall or not, the proc above needs to be called from another wall or it won't work. I believe it's designed like this for performance

Reply:
Also there's some chunk of code on the Destroy() proc that makes me think near walls had to update after dismantling, but it never worked

Reply:
If an object is dismantled, it doesn't need to update the smoothing of objects it can't smooth with, because its destruction can never change how they should be smoothed. Thus, those performance checks should never prevent smoothing from updating.

Reply:
I just checked and _ChangeTurf_ resets the _canSmoothWith_ list, so the proc _relativewall_neighbours_ is not useful here

Reply:
Alright, I see the problem, and am also an idiot
That seems like a pretty big oversight in the smoothwall system, honestly

This could still pose a significant performance issue in explosions, though
Actually, let me test it out

Reply:
Surprisingly, it's actually fine
According to profiling on the PR test server, the time spent processing this proc during a maxcap explosion is slightly greater than the processing time required to spawn one (1) `/mob/living`